packages/date-picker/src/vaadin-date-picker-overlay-content-mixin.js

```diff
@@ -1009,18 +1028,23 @@ export const DatePickerOverlayContentMixin = (superClass) =>
     }
 
     /** @private */
-    _dateAllowed(date, min = this.minDate, max = this.maxDate) {
-      return (!min || date >= min) && (!max || date <= max);
+    _dateAllowed(date, min = this.minDate, max = this.maxDate, isDateDisabled = this.isDateDisabled) {
+      let dateIsDisabled = false;
+      if (typeof isDateDisabled === 'function' && !!date) {
+        const dateToCheck = extractDateParts(date);
+        dateIsDisabled = isDateDisabled(dateToCheck);
+      }
+      return (!min || date >= min) && (!max || date <= max) && !dateIsDisabled;
```

Comment:
Would it be an option to use the `dateAllowed` from `-helper.js` here instead of duplicating the logic?

Reply:
That makes perfect sense. It is identical logic. The tests all pass and funcionality works just fine.